### DISCUSSION TITLE
Breaking out of loop / while with encoder

### DISCUSSION THREAD

**Discussion by kimbrussow** [forntoh/LcdMenu#271] [Q&A]
I am using LcdMenu to control processing (led light strip / FastLED)
one of my SubMenus has the following:
ITEM_COMMAND("Pattern 1", []() {
    Pat1();
    }),  << calls the function:
    void Pat1()
{ 
do{
       for (int i = 0; i < NUM_LEDS; i++) {
// for testing loop      Serial.println("in loop");
       leds[i] = CRGB::Yellow;
       FastLED.show();
      delay(100);
    blah blah blah...
    } while("get something from the encoder here"!= true);
}
Question: how to "get something from the encoder" to exit the do loop??

**Comment by B1ackt34** [comment]
Read here
https://github.com/forntoh/LcdMenu/discussions/267#discussioncomment-11359946
